### PR TITLE
simulators/ethereum/engine: Add incorrect engine versions tests

### PR DESCRIPTION
**Pull request by marioevz**
Addresses missing GetPayloadV2/V3 and ForkchoiceUpdatedV2/V3 tests at fork time, described in comment: https://github.com/ethereum/hive/issues/810#issuecomment-1672182427

### CODE REVIEW

**Comment by marioevz** [comment]
Hi @jochem-brouwer @g11tech, I tried to add the test described in the ACD on commit https://github.com/ethereum/hive/pull/834/commits/0b0390dd6450c8101c957ee8f3fb429443b531b8, could you let me know if this addresses the edge case? Thanks!

**Comment by g11tech** [comment]
> Hi @jochem-brouwer @g11tech, I tried to add the test described in the ACD on commit https://github.com/ethereum/hive/pull/834/commits/0b0390dd6450c8101c957ee8f3fb429443b531b8, could you let me know if this addresses the edge case? Thanks!

Will check and update 👍

File: simulators/ethereum/engine/suites/cancun/tests.go [hunk 1]
```diff
@@ -537,6 +537,235 @@ var Tests = []test.SpecInterface{
...
+	},
+
+	// ForkchoiceUpdatedV2 after cancun
+	&CancunBaseSpec{
+		Spec: test.Spec{
+			Name: "ForkchoiceUpdatedV2 To Request Cancun Payload, Zero Beacon Root",
+			About: `
+			Test sending ForkchoiceUpdatedV2 to request a Cancun payload:
+			- Payload Attributes uses Cancun timestamp
+			- Payload Attributes' Beacon Root zero
```
Comment (g11tech) [inline simulators/ethereum/engine/suites/cancun/tests.go:665]: would like to avoid interpreting beacon root (check against zero) to return error. not essential imo but also no strong opinions here

**Review by g11tech — CHANGES_REQUESTED** [review]
@marioevz mostly looks amazing

the test case which i was requesting on the call was:

1. send fcUv3 with payload attributes with a parent beacon block root `x`
2. resenf fcUv3  with payload attributes with updated parent beacon block root `y` (lets say CL changed its head) but rest everything same

checks:

1. the payload id for 1 & 2 should be different,
2.  fetch the payload for new payload id recieved in 2nd fcUv3 request 
3. new payload v3 request with the new payload and the updated beacon block root `y` should return `VALID`

File: simulators/ethereum/engine/suites/cancun/tests.go (hunk 1 above)
Comment (marioevz) [reply simulators/ethereum/engine/suites/cancun/tests.go:665]: My intention was to test that the client responds "Invalid Params" when the beacon root is provided for FcUv2, which should not contain a beacon root, trying to follow this line on v3's spec: "provided set of parameters and their fields strictly matches the expected one".

The zero beacon root was just to represent a non-nil value, any non-nil value should (in my opinion) result in the same error.

**Comment by marioevz** [comment]
@g11tech I merged the new test and verified before and after EthJS fix and it correctly detected the issue/passes the test on fix!

File: simulators/ethereum/engine/suites/cancun/tests.go (hunk 1 above)
Comment (g11tech) [reply simulators/ethereum/engine/suites/cancun/tests.go:665]: ahh ok thanks for clarification
